Applications/Terminal/Terminal.cpp

```diff
@@ -457,20 +558,34 @@ void Terminal::execute_xterm_command()
 
 void Terminal::execute_escape_sequence(byte final)
 {
+    m_question_param = false;
     m_final = final;
-    auto paramparts = String::copy(m_parameters).split(';');
     ParamVector params;
+
+    if (m_parameters.size() > 0 && m_parameters[0] == '?') {
+        m_question_param = true;
+        m_parameters.remove(0);
+    }
+    auto paramparts = String::copy(m_parameters).split(';');
     for (auto& parampart : paramparts) {
         bool ok;
         unsigned value = parampart.to_uint(ok);
         if (!ok) {
+            // FIXME: Should we do something else?
             m_parameters.clear_with_capacity();
             m_intermediates.clear_with_capacity();
-            // FIXME: Should we do something else?
             return;
         }
         params.append(value);
     }
+
+    dbgprintf("Terminal::execute_escape_sequence: Handled final '%c'\n", final);
```

Comment:
This is going to be pretty noisy. I'd suggest sticking it in an:

    #if defined(TERMINAL_DEBUG)
    #endif